### DISCUSSION TITLE
Crash at initial import fitz

### DISCUSSION THREAD

**Discussion by macdeport** [pymupdf/PyMuPDF#1302] [Looking for help] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
When executing initial import:

```
Traceback (most recent call last):
  File "/var/folders/0d/dj43zrpj3kg85hdt91_cjwy00000gp/T/BBEditRunTemp-untitled text 38.py", line 1, in <module>
    import fitz
  File "/Users/alain/Library/Python/3.9/lib/python/site-packages/fitz/__init__.py", line 10, in <module>
    from fitz.fitz import *
  File "/Users/alain/Library/Python/3.9/lib/python/site-packages/fitz/fitz.py", line 15, in <module>
    from . import _fitz
ImportError: dlopen(/Users/alain/Library/Python/3.9/lib/python/site-packages/fitz/_fitz.cpython-39-darwin.so, 2): Symbol not found: ____chkstk_darwin
  Referenced from: /Users/alain/Library/Python/3.9/lib/python/site-packages/fitz/_fitz.cpython-39-darwin.so
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/alain/Library/Python/3.9/lib/python/site-packages/fitz/_fitz.cpython-39-darwin.so

```

## To Reproduce (mandatory)
Explain the steps to reproduce the behavior, For example, include a minimal code snippet, example files, etc.
My installation of PyMuPDF:
` pip-3.9 install -U pymupdf`

## Expected behavior (optional)
Describe what you expected to happen (if not obvious).

## Your configuration (mandatory)
 - Operating system, potentially version and bitness
 - Python version, bitness
 - PyMuPDF version, installation method (**wheel** or **generated** from source):
 **wheel** method

For example, the output of `print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` would be sufficient (for the first two bullets).

```
3.9.7 (default, Sep  1 2021, 02:08:24) 
[Clang 7.0.2 (clang-700.1.81)] 
darwin  
```

**Comment by JorjMcKie** [comment]
Well, obviously incorrectly installed.
This is an Intel machine, right?
I have no Mac, so cannot reproduce the error. What other information do you have?
How does your pymupdf Python installation directory look like? I.e. this one: `/Users/alain/Library/Python/3.9/lib/python/site-packages/fitz`
What was the console protocol of the installation?

**Comment by macdeport** [comment]
First of all, thank you for trying to help me quickly.

> Well, obviously incorrectly installed.

Yes

> This is an Intel machine, right?

Yes

> I have no Mac, so cannot reproduce the error. What other information do you have?

A quite "old" system => _Mac OS X 10.10.5 (Yosemite)_

> How does your pymupdf Python installation directory look like? I.e. this one: 
> `/Users/alain/Library/Python/3.9/lib/python/site-packages/fitz`

![floating-screenshot-184802](https://user-images.githubusercontent.com/26068595/135132582-b17f388e-24bf-4ea5-96c4-969aeca4eefd.jpg)

![floating-screenshot-190951](https://user-images.githubusercontent.com/26068595/135133798-45a01d51-feaa-401c-95f6-1f9a93694b53.jpg)

What was the console protocol of the installation?
![floating-screenshot-191240](https://user-images.githubusercontent.com/26068595/135134061-ab140e6d-70fb-40ba-8bae-5601adc346e5.jpg)

You are welcome, thanks,

**Comment by JorjMcKie** [comment]
Weird. I just made a little Github action to test installation and invocation of pymupdf: it did work, here is the log
[1_Install on OSX.txt](https://github.com/pymupdf/PyMuPDF/files/7246744/1_Install.on.OSX.txt)

**Comment by JorjMcKie** [comment]
As a sort of last resort: try to upgrade pip

**Comment by macdeport** [comment]
https://github.com/pymupdf/PyMuPDF/issues/1296#issuecomment-929552869
@JorjMcKie  If I understand correctly: from the log you shared I discover that you can follow the progress of my pip installation of PyMuPDF.

This installation seems to have gone smoothly.

But on closer inspection there seems to be an inconsistency, still if I understand well: the MacOS 10.15.7 version is invoked while I (still) use the 10.10.5 version :-(

![floating-screenshot-225230](https://user-images.githubusercontent.com/26068595/135167900-71cb2be0-d166-4e15-a871-dfbc23127212.jpg)

The pip install script seems to confuse 10.15.7 <=> 10.10.5 and install the wrong version of the binary 
_fitz.cpython-39-darwin.so

Under these conditions is it realistic to try the "generated from source" method unless you have other versions of '_fitz.cpython-39-darwin.so'?

**Comment by macdeport** [comment]
> As a sort of last resort: try to upgrade pip

Unfortunately I have already try to updated pip:
![floating-screenshot-233308](https://user-images.githubusercontent.com/26068595/135168849-4cd1739e-1004-4036-92fb-70b0807a4db2.jpg)

**Comment by JorjMcKie** [comment]
> Under these conditions is it realistic to try the "generated from source" method unless you have other versions of '_fitz.cpython-39-darwin.so'?

Yes, I would do that! And no, this is the only OSX stuff I am able to generate.
If you need an installation script for this undertaking, you may want to derive it from [here](https://github.com/pymupdf/PyMuPDF/blob/master/installation/freebsd/freebsd_pymupdf.sh).

**Comment by macdeport** [comment]
So I will try. Thank you for your documented help.

**Comment by mlove4u** [comment, marked as answer]
In my Case:
Mac OS: 10.12.1
Python3: 3.7.4

When pymupdf<=1.18.14, it can be imported. But error when version>=1.18.15.
![versions](https://user-images.githubusercontent.com/18019314/147062867-b71c6d5d-86e3-464a-9b60-d86f6970eb48.png)

**Reply by macdeport** [reply]
I can confirm on Mac OS: 10.10.15

**Comment by G3zz** [comment]
I am still hitting this issue - did anyone manage to resolve and get 1.18.17 working? I'm on an m1 machine.
EDIT: https://github.com/pymupdf/PyMuPDF/issues/429 following advice from here I double checked that pymupdf was the same version as the installed mupdf and it was not. Once I matched them (1.19.1 in my case), the import started working 😄
